plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -108,12 +133,14 @@ public void run() {
             try {
                 while (running) {
                     try {
-                        Thread.sleep(3000);
-                        updatePipelines();
+                        Thread.sleep(pipelineUpdateInterval.millis());
```

Comment:
The updater is based on the `TTLService#PurgeThread`, personally I don't see a big harm in have a thread that is always around for checking updates. I have no objections to change it to be based on scheduled executor.